internal/cloudinfo/machine_types.go

```diff
@@ -54,14 +56,38 @@ type VMKey struct {
 	cloud        string
 	service      string
 	region       string
-	location     string
 	instanceType string
 }
 
+type InstanceTypeMap struct {
+	sync.RWMutex
+	internal map[VMKey]MachineDetails
+}
+
+func NewInstanceTypeMap() *InstanceTypeMap {
+	return &InstanceTypeMap{
+		internal: make(map[VMKey]MachineDetails),
+	}
+}
+
+func (im *InstanceTypeMap) getVM(key VMKey) (MachineDetails, bool) {
+	im.RLock()
+	result, ok := im.internal[key]
+	im.RUnlock()
+	return result, ok
+}
+
+// nolint: gochecknoglobals
+var instanceTypeMap = NewInstanceTypeMap()
+
 // nolint: gochecknoglobals
-var instanceTypeMap = make(map[VMKey]MachineDetails)
+var log logrus.FieldLogger
 
-func fetchMachineTypes(cloud string, service string, region string, location string) error {
+func init() {
+	log = config.Logger()
+}
+
+func fetchMachineTypes(cloud string, service string, region string) error {
```

Comment:
Can we inject `cloudInfoEndPoint` as parameter instead of `viper.GetString(config.CloudInfoEndPoint)`?